### PR TITLE
Reorganize reference section and add billing documentation

### PR DESCRIPTION
**Pull request by hasheddan**
Signficantly reorganizes the reference section of the documentation to reflect new concepts and remove stale information. Also introduces a new section on billing and usage data.

### CODE REVIEW

File: docs/reference/3-management-api/README.md [hunk 1]
```diff
@@ -14,16 +13,14 @@ export const OpenAPIUrl = () => {
   return <a href={openAPIUrl}>OpenAPI v3</a>;
 };
 
-You can use Golioth API for access all services available on the platform, like
-device management, provisioning, data resources and others services to be build
-in the future.
+All Golioth platform operations can be performed via the Management API.
 
 #### API Usage
```
Comment (szczys) [inline docs/reference/3-management-api/README.md:18]:
```suggestion
## API Usage
```

File: docs/reference/3-management-api/README.md [hunk 2]
```diff
@@ -14,16 +13,14 @@ export const OpenAPIUrl = () => {
...
-Our API is documented using OpenAPI definitions. You can access the [API
-Reference Page](/reference/rest-api/openapi) to check all endpoints and methods
-that can be used. For more information on authentication check the
-[Authentication Guide](/reference/rest-api/auth).
+The API is documented using OpenAPI definitions. You can access the [API
+Reference Page](/reference/management-api/openapi) to check all endpoints and
+methods that can be used. For more information on authentication check the
+[Authentication Guide](/reference/management-api/auth).
 
 #### Importing OpenAPI/Swagger Definitions
```
Comment (szczys) [inline docs/reference/3-management-api/README.md:25]:
```suggestion
## Importing OpenAPI/Swagger Definitions
```

File: docs/reference/6-billing/1-usage.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
+---
+title: Usage
+---
+
+A subset services offered by the Golioth platform are billed based on monthly
```
Comment (szczys) [inline docs/reference/6-billing/1-usage.md:5]:
```suggestion
A subset of services offered by the Golioth platform are billed based on monthly
```

File: docs/reference/6-billing/1-usage.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+usage. These services are identified in documentation via the following.
```
Comment (szczys) [inline docs/reference/6-billing/1-usage.md:6]:
```suggestion
usage. These services are identified in documentation using notification cards like the one shown below.
```

**Review by szczys — APPROVED** [review]
This reorg looks great, nice job!

It would be good to add a README.md in the API Docs folder so that it is possible to directly link to this section. This can be as simple as a page that automatically generates cards for each subpage: https://training.golioth.io/docs/zephyr-training/helloworld

I'm happy to open a PR for this if that is helpful, just let me know.

**Review by hasheddan — COMMENTED** [review]
> It would be good to add a README.md in the API Docs folder so that it is possible to directly link to this section. This can be as simple as a page that automatically generates cards for each subpage: https://training.golioth.io/docs/zephyr-training/helloworld

> I'm happy to open a PR for this if that is helpful, just let me know.

Thanks for this pointer @szczys! Added 👍🏻

**Pull request merged**